### PR TITLE
feat: add user avatar

### PR DESCRIPTION
**Pull request by gondar00**
fixes: #3018 

Deployed on: https://useravatar.herokuapp.com

### CODE REVIEW

File: frontend/src/SettingsPage/SettingsPage.jsx [hunk 1]
```diff
@@ -101,16 +116,35 @@ function SettingsPage(props) {
...
+                      <input type="text" className="form-control" name="email" value={email} readOnly disabled />
+                    </div>
+                  </div>
+                  <div className="col">
+                    <div className="mb-3">
+                      <div className="form-label">Avatar</div>
+                      <input
+                        onChange={(e) => {
+                          const file = e.target.files[0];
+                          console.log(file);
```
Comment (akshaysasidrn) [inline frontend/src/SettingsPage/SettingsPage.jsx:133]: missed to remove

File: server/src/controllers/files.controller.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+} from '@nestjs/common';
+import { Readable } from 'stream';
+import { Response } from 'express';
+import { FilesService } from '../services/files.service';
+import { UpdateFileDto } from '../dto/update-file.dto';
+import { JwtAuthGuard } from 'src/modules/auth/jwt-auth.guard';
+
+@Controller('files')
+@UseInterceptors(ClassSerializerInterceptor)
+export class FilesController {
```
Comment (akshaysasidrn) [inline server/src/controllers/files.controller.ts:21]: If we aren't using `#findAll` and `#update`, could we remove them?

Comment (akshaysasidrn) [reply server/src/controllers/files.controller.ts:21]: Also, let's try to stick with current naming conventions ie #index, #show etc when possible within a controller?

File: server/src/controllers/users.controller.ts [hunk 3]
```diff
@@ -37,6 +48,14 @@ export class UsersController {
     };
   }
 
+  @Post('avatar')
+  @UseGuards(JwtAuthGuard)
+  @UseInterceptors(FileInterceptor('file'))
+  async addAvatar(@Req() req, @UploadedFile() file: Express.Multer.File) {
+    if (!file || !file?.buffer) throw new BadRequestException('File cannot be empty');
```
Comment (akshaysasidrn) [inline server/src/controllers/users.controller.ts:55]: could use dto directly for this to validate here and then pass it onto the service

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]

File: server/src/controllers/files.controller.ts (hunk 2 above)
Comment (gondar00) [reply server/src/controllers/files.controller.ts:21]: Cool, I have removed the unused endpoints and renamed the fn

**Review by akshaysasidrn — APPROVED** [review]

**Review by akshaysasidrn — APPROVED** [review]

**Pull request merged**